### PR TITLE
enable flat fee option on swaps

### PR DESCRIPTION
**Pull request by ewilz**
it can be difficult to use PAY_PORTION with exactOut trades to be able to get a clean output amount for the swapper, since we're always taking a percentage of total amount out (exact + fee %). So we must work backwards from that total amount, and since bips don't have very high precision, it becomes very hard to get the swapper their exactOut wihtout imprecision.

This PR allows taking a flat fee to accomplish clean numbers on exactOut trades.

### CODE REVIEW

File: src/entities/protocols/uniswap.ts [hunk 1]
```diff
@@ -107,6 +115,25 @@ export class UniswapTrade implements Command {
         }
       }
 
+      // If there is a flat fee, that absolute amount is sent to the fee recipient
+      // In the case where ETH is the output currency, the fee is taken in WETH (for gas reasons)
```
Comment (zhongeric) [inline src/entities/protocols/uniswap.ts:119]: hmm, so the fee transfer happens after the swap but before the unwrap?

File: test/uniswapTrades.test.ts [hunk 2]
```diff
@@ -251,6 +251,23 @@ describe('Uniswap', () => {
       expect(methodParameters.value).to.eq(methodParametersV2.value)
     })
 
+    it('encodes a single exactOutput ETH->USDC swap, with a flat fee', async () => {
```
Comment (zhongeric) [inline test/uniswapTrades.test.ts:254]: maybe can add an exactOut, eth out test since we have that edge case where the fee is taken in WETH just to make sure its working with unwrap logic

**Review by zhongeric — APPROVED** [review]
